pennylane_lightning/lightning_qubit.py

```diff
@@ -131,6 +145,62 @@ def apply_lightning(self, state, operations):
 
         return np.reshape(state_vector, state.shape)
 
+    def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
+        if not CPP_BINARY_AVAILABLE:
+            return super().adjoint_jacobian(tape, starting_state, use_device_state)
+
+        if self.shots is not None:
+            warn(
+                "Requested adjoint differentiation to be computed with finite shots."
+                " The derivative is always exact when using the adjoint differentiation method.",
+                UserWarning,
+            )
+
+        if len(tape.trainable_params) == 0:
+            return np.array(0)
+
+        for m in tape.measurements:
+            if m.return_type is not Expectation:
+                raise QuantumFunctionError(
+                    "Adjoint differentiation method does not support"
+                    f" measurement {m.return_type.value}"
+                )
+
+        for op in tape.operations:
+            if op.num_params > 1 and not isinstance(op, Rot):
+                raise QuantumFunctionError(
+                    f"The {op.name} operation is not supported using "
+                    'the "adjoint" differentiation method'
+                )
```

Comment:
Another flyby comment just to ask why the other parametrized ops (RX, RY, RZ, PhaseShift, etc.) are not supported? Is that coming in a later PR?

Reply:
Nevermind, I missed the `op.num_params > 1` 🤦